app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/ExposureNotificationsRepository.kt

```diff
@@ -311,14 +311,23 @@ class ExposureNotificationsRepository @Inject constructor(
                 accepted = false
             )
         })
-        val latestExposure = db.dao().getLatest().firstOrNull()?.daysSinceEpoch
-        val lastNotifiedExposure = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
-        if (latestExposure != null && latestExposure != lastNotifiedExposure) {
+
+        val latestExposure = db.dao().getLatest().firstOrNull()
+        val latestExposureTime = latestExposure?.daysSinceEpoch
+        val lastNotifiedExposureTime = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
+        val latestExposureTimestamp = latestExposure?.importTimestamp ?: 0L
+
+        val appOpenedTimestamp = prefs.getLastTimeAppOpened()
+        val appOpenedAfterLastNotification = latestExposureTimestamp < appOpenedTimestamp
+        val shouldShowNotification =
+            latestExposureTime != lastNotifiedExposureTime || !appOpenedAfterLastNotification
+
+        if (latestExposure != null && shouldShowNotification) {
```

Comment:
Dej ten nullcheck taky do té boolean proměnné `shouldShowNotification`, takhle to je matoucí

Reply:
zrefaktorovala som celu metodu a pridala som komentare